src/inmanta/server/services/environment_metrics_service.py

```diff
@@ -243,3 +244,37 @@ async def get_metric_value(
             assert isinstance(record["status"], str)
             metric_values.append(MetricValue(self.get_metric_name(), record["count"], record["environment"], record["status"]))
         return metric_values
+
+
+class AgentCountMetricsCollector(MetricsCollector):
+    """
+    This Metric will track the number of agents (grouped by agent status).
+    """
+
+    def get_metric_name(self) -> str:
+        return "resource.agent_count"
```

Comment:
this https://github.com/inmanta/inmanta-core/blob/master/src/inmanta/server/services/environment_metrics_service.py#L63
prevents me form changing the name to `resource.agent_count`. I'm not really sure what I should do here? something like this?

```
    def __init__(self, target: string, metric_name: str, count: int, environment: uuid.UUID, grouped_by: Optional[str] = None) -> None:
        if METRIC_NAME_SEPARATOR in metric_name:
            raise Exception('The character "%s" can not be used in the metric_name (%s)' % (METRIC_NAME_SEPARATOR, metric_name))
        if METRIC_NAME_SEPARATOR in target:
            raise Exception('The character "%s" can not be used in the target (%s)' % (METRIC_NAME_SEPARATOR, target))
        if grouped_by and METRIC_NAME_SEPARATOR in grouped_by:
            raise Exception(
                'The character "%s" can not be used in the grouped_by value (%s)' % (METRIC_NAME_SEPARATOR, grouped_by)
            )
        self.metric_name = ".".join([target, metric_name, grouped_by]) if grouped_by else  ".".join([target, metric_name]) 
        self.count = count
        self.environment = environment
```
and add `resource` via `target`? 
or diverge from the design and use an underscore `resource_agent_count or some other separator?`

Reply:
Why not use a another separator characters like `#` for example? That way it's clear that the first part is the name and the second part the subdivision.

Reply:
that works for me to :) I'll do that